src/components/DatePicker.tsx

```diff
@@ -1,32 +1,24 @@
 import React from "react";
-import DayPicker, { NavbarElementProps, WeekdayElementProps, Modifier } from "react-day-picker";
-import { OverlayTriggerState } from "react-stately";
+import DayPicker, { Modifier, NavbarElementProps, WeekdayElementProps } from "react-day-picker";
 import { IconButton } from "src/components";
 import { Css, Palette } from "src/Css";
 import { useTestIds } from "src/utils";
+import "./DatePicker.css";
 
-interface DatePickerOverlayProps {
-  value: Date | undefined;
-  state: OverlayTriggerState;
-  positionProps: React.HTMLAttributes<Element>;
-  onChange: (value: Date) => void;
-  disabledDays: Modifier;
+export interface DatePickerProps {
+  value?: Date;
+  onSelect: (value: Date) => void;
+  disabledDays?: Modifier;
 }
 
-export function DatePickerOverlay(props: DatePickerOverlayProps) {
-  const { value, state, positionProps, onChange, disabledDays } = props;
-  // We define some spacing between the Calendar overlay and the trigger element, and depending on where the overlay renders (above or below) we need to adjust the spacing.
-  // We can determine if the position was flipped based on what style is defined, `top` (for positioned below the trigger), and `bottom` (for above the trigger).
-  // The above assumption regarding `top` and `bottom` is true as long as we use `bottom` as our default `OverlayPosition.placement` (set in DateField).
-  // The reason the placement may not be on bottom even though we set `bottom` is because also set `shouldFlip: true`
-  const isPositionedAbove = !positionProps.style?.top;
+export function DatePicker(props: DatePickerProps) {
```

Comment:
Maybe include a jsdoc like "Renders a DatePicker inline in the current DOM, see ButtonDatePicker and DateField."

...do we expect users to use this `src/components/DatePicker`? Fwiw after pulling up `ButtonDatePicker` in storybook, I almost wonder if that component should be the "public `DatePicker` component", and this one is an internal `InlineDatePicker`?

Or maybe not, I'm not really sure, so feel free to leave as-is if this `src/components/DatePicker` is actually a useful/standalone public component.

Reply:
My gut feeling is that it might be worth being a standalone component. Will move it to be an internal component for now.